pkg/apis/navigator/validation/elasticsearch.go

```diff
@@ -102,6 +103,32 @@ func ValidateElasticsearchCluster(esc *navigator.ElasticsearchCluster) field.Err
 	return allErrs
 }
 
+func ValidateElasticsearchClusterUpdate(old, new *navigator.ElasticsearchCluster) field.ErrorList {
+	allErrs := ValidateElasticsearchCluster(new)
+
+	fldPath := field.NewPath("spec")
+
+	npPath := fldPath.Child("nodePools")
+	for i, newNp := range new.Spec.NodePools {
+		idxPath := npPath.Index(i)
+
+		for _, oldNp := range old.Spec.NodePools {
+			if newNp.Name == oldNp.Name {
+				if !reflect.DeepEqual(newNp.NodeSelector, oldNp.NodeSelector) {
+					allErrs = append(allErrs, field.Forbidden(idxPath.Child("nodeSelector"), "cannot modify nodeSelector"))
```

Comment:
Why do we not allow modifying the nodeSelector? It's an updatable field in StatefulSet. If there are issues around how it plays with the `rack` parameter/node selector targeting a particular zone, we may need to rethink what we're doing there.

Reply:
We'd need an associated action to manipulate the nodeSelector - currently we only modify replicas.